Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -472,9 +474,9 @@ void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
 
   // Create the peak using the Q in the lab framewith all its info:
   Peak p(m_inst, q);
-  int edge = this->getProperty("EdgePixels");
-  if (edge > 0) {
-    if (edgePixel(p.getBankName(), p.getCol(), p.getRow(), edge))
+  if (m_edge > 0) {
+    if (DetectorInfo::edgePixel(m_inst, p.getBankName(), p.getCol(), p.getRow(),
```

Comment:
How does this actually impact on the performance of `PredictPeaks`? It's something I did not immediately think of on the original PR that introduced this, but it might actually be quite expensive to enable this, since it does instrument tree traversal etc. internally.

Is it worth thinking about generalising this further, simply so that masked detectors are rejected? Or maybe they are already ignored somewhere further down in the ray-tracing code?

I think that would make it both faster and more flexible, because the mask can really come from anywhere. It would have to be created once for an instrument and could then be reused. Maybe somebody who is more familiar with masking can comment on the pitfalls of that approach.